### ISSUE TITLE
Add functionality to retrieve decision boundary

### ISSUE THREAD

**Issue by tlienart** [alan-turing-institute/MLJ.jl#342]
Use cases are typically visualisation but could also be further analysis like robustness (via margins) etc.

For some models they should be very easy  to  get (SVM, linear classifiers, LDA, ...) though sometimes dependent upon a threshold. 

It's hard in general to describe the decision boundary but for the basic models we should provide a way.

**Comment by ablaom** [comment]
I suggest `report` as the interface point.

**Comment by tlienart** [comment]
yes that makes sense, so would you have a  named field with a callable?

**Comment by ablaom** [comment]
Yes. The user does something like `report(mach).get_boundary(threshold)` or whatever. 

I hindsight `extras` would have been a better name that `report`. I guess it's too late for that now.

**Comment by ablaom** [comment]
Or though we could switch and keep `report` for backwards compatibility.
